speedy-recovery-client/src/components/secured/core/SecuredMainPage.js

```diff
@@ -85,10 +85,23 @@ class SecuredMainPage extends Component {
   }
 
   updateStateAppointments = () => {
-    // TODO: Query SMART, display appointment data for current user
-    const appointments = fhirExampleApp.map((app) => fhirMapAppointment(app, this.props.fhirVersion));
+    FhirDataQueryingService.getUserAppointment(this.props.user.id)
+      .then(appointment => {
+        const appointments = this.handleAppointment(appointment);
+        this.setState({ appointments });
+      }).catch(errorMessage=>{this.setState([])});
+  };
 
-    this.setState({ appointments });
+  handleAppointment = appointment => {
+    const appointments = [];
+    if(appointment.data.total !== 0){
+      for (var i = 0; i < appointment.data.entry.length; i++){
```

Comment:
This could be shortened to something like `appointment.data.entry.map((app)` => fhirMapAppointment(app.resource, this.props.fhirVersion))`